### PR TITLE
[spirv] Vectorize integer extend ops in lowering to subgroup_mma

### PR DESCRIPTION
**Pull request by qedawkins**
For integer types, integer extend ops are matched against neighboring vector.transfer_read/contract ops when lowering to mma ops. This enables vectorizing the extend ops to cooperative matrix sizes. This also enables support for cases with mixed signedness.

Depends on https://reviews.llvm.org/D143922

### CODE REVIEW

**Review by antiagainst — APPROVED** [review]
Nice! The upstream patch this pull request depends on should be in now.

**Comment by qedawkins** [comment]
Yay, I cannot land this though. Do you want me to rebase and rerun the CI to check before landing?

**Pull request merged**
